### PR TITLE
Add shared channels related configuration under a E20 license check

### PR DESCRIPTION
**Pull request by nronas**
#### Summary
Add's shared channels and remote_cluster_service on license features.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-29027
https://mattermost.atlassian.net/browse/MM-30838

#### Ref PRs: 
Redux for adding the new config types: https://github.com/mattermost/mattermost-redux/pull/1305
Webapp for adding the setting entry on the admin console: https://github.com/mattermost/mattermost-webapp/pull/7136/
License generator: https://github.com/mattermost/license-generator/pull/13

#### Release Note
```release-note
Add's shared channels and remote_cluster_service on license features.
```

**NOTE**: This PR has been opened against master and not against the feature branch, since the initial configuration for shared channels were already in master.

### CODE REVIEW

**Review by marianunez — CHANGES_REQUESTED** [review]

**Review by agnivade — APPROVED** [review]
Nothing to add on top of Maria's review. Good to go from my side after that.

File: model/license.go
Comment (nronas) [reply model/license.go:87]: Yeap, it was on my TODO list but i completely missed it :). 👍

File: config/client.go
Comment (nronas) [reply config/client.go:205]: completely miss the section above that we default all keys in the config. Thanks for the suggestion.

File: model/license.go
Comment (nronas) [reply model/license.go:237]: My understanding is that RemoteClusterService should not be configured without SharedChannels being configured. I will refer to @wiggin77 for asserting my assumption for this, in the meanwhile will guard against RemoteClusterService being nil.

Comment (nronas) [reply model/license.go:237]: Now that I am thinking it more, `SharedChannels` cannot be nil since we set a default value to it before we set the `RemoteClusterService` default. So we should be fine on this.

File: model/license.go
Comment (nronas) [reply model/license.go:87]: Amended in https://github.com/mattermost/license-generator/pull/13

**Review by jwilander — APPROVED** [review]

**Review by agnivade — APPROVED** [review]
@nronas - You already have my approval :) The convention we use is that once an approver approves with some suggested changes, you don't need to re-request a review again unless you explicitly want the reviewer to take a look at your code again.

File: model/license.go
Comment (marianunez) [reply model/license.go:237]: Good point! Complete overlook from my side 👍

**Review by marianunez — APPROVED** [review]
LGTM

File: model/license.go
Comment (wiggin77) [reply model/license.go:237]: >  My understanding is that RemoteClusterService should not be configured without SharedChannels being configured. 

@nronas just for clarity, the Shared Channels service relies on Remote Cluster service for comms, therefore if Shared Channels is enabled, Remote Clusters must be as well.  Post MVP we'll create separate UI for enabling Remote Clusters independent of Shared Channels since other things will be shared/communicated between clusters.

**Review by wiggin77 — APPROVED** [review]
LGTM  :+1:

**Comment by nronas** [comment]
### Tested on Test Server

Works as expected:

Screenshot 2020-12-03 at 12 00 53 PM

**Comment by nronas** [comment]
### Acceptance Tests

- Server Should load without any issues
- Config should include under `ExperimentalSettings` an entry called `EnableSharedChannels` that is either false for non-license and E10 license but respects the value of that Feature that is defined in an E20 license

**Comment by wiggin77** [comment]
@nronas are you ready to merge this now?

**Comment by nronas** [comment]
@wiggin77 i was waiting for QA to give me the go-ahead on this. I've done some testing on my side, all is looking well.

**Comment by wiggin77** [comment]
> @wiggin77 i was waiting for QA to give me the go-ahead on this. I've done some testing on my side, all is looking well.

Ok, got it.  The Reviews Complete tag should be removed.

**Comment by nronas** [comment]
> Ok, got it. The Reviews Complete tag should be removed.

Oh I thought the reviews complete was for code review only, awesome, noted.

**Comment by lindalumitchell** [comment]
Thanks @nronas! I see there are a couple other PRs related to this; I'd prefer to take a look at a test server when the feature appears in System Console (doesn't appear yet on this PR without the other(s). Is that possible, maybe for me to review this on one of the other PRs? And we can wave this one through in favor of me reviewing on the other one(s)?

**Comment by nronas** [comment]
👋 @lindalumitchell pretty new here, I do not know if we can have a test server with webapp + server on specific shas, let me ask around to see what I can find.

**Comment by nronas** [comment]
So @lindalumitchell i will go ahead and merge this to master and you can test that by spinning up a test server on the webapp PR https://github.com/mattermost/mattermost-webapp/pull/7136

File: model/license.go
Comment (jwilander) [inline model/license.go:233]: @nronas if you want this to be enabled for E20 license you should set this to `f.FutureFeatures`. Otherwise old E20 licenses will not get this new feature unless we re-issue them a new license

**Pull request merged**
